app/services/set_decline_by_default.rb

```diff
@@ -14,6 +14,7 @@ def call
     # applications have either an offer, been rejected by provider, withdrawn by
     # the candidate, or have the offer withdrawn.
     final_decision_date = [
```

Comment:
Can we add a spec that covers this behaviour?

Reply:
We are already have some coverage for this in https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/spec/services/set_decline_by_default_spec.rb#L153 but I've modified the 'offer updated' scenario to use `offer_changed_at`.